cachier/pickle_core.py

```diff
@@ -186,13 +247,18 @@ def mark_entry_not_calculated(self, key):
                 pass  # that's ok, we don't need an entry in that case
 
     def wait_on_entry_calc(self, key):
-        with self.lock:
-            self._reload_cache()
-            entry = self._get_cache()[key]
-            if not entry['being_calculated']:
-                return entry['value']
+        if self.separate_files:
+            entry = self._get_cache_by_key(key)
+            filename = f'{self._cache_fname()}_{hashlib.sha256(pickle.dumps(key)).hexdigest()}'
+        else:
+            with self.lock:
+                self._reload_cache()
+                entry = self._get_cache()[key]
+            filename = self._cache_fname()
+        if not entry['being_calculated']:
```

Comment:
I don't get it. Your `_get_cache_by_key` can definitely return `None`. Doesn't this yields a `KeyError` on line 258?

Reply:
Truth is I'm a bit confused regarding what happens when the cache file is corrupt. I can return {'value': None} instead of None and the problem will be "fixed" but I'm not sure that I handle stuff correctly when cache files become corrupt. I'll try to understand the test you wrote better and create a seperate_files version.

Reply:
OK, it'll return `None` only if the file doesn't exist, and then why would we ever enter `_wait_on_entry_calc`?

Reply:
What do you mean? The call to `wait_on_entry_calc` is unrelated, and is done by the generic code if it encounters `entry['being_calculated'] == True`. This can happen for you code with two fast consecutive calls with the same argument set.